src/hex/rings.rs

```diff
@@ -175,11 +179,11 @@ impl Hex {
     /// If you only need the coordinates see [`Self::wedge`]
     pub fn custom_ring_edges(
         self,
-        range: impl Iterator<Item = u32>,
+        ranges: impl Iterator<Item = u32>,
         direction: DiagonalDirection,
         clockwise: bool,
     ) -> impl Iterator<Item = Vec<Self>> {
-        range.map(move |r| self.custom_ring_edge(r, direction, clockwise))
+        ranges.map(move |r| self.custom_ring_edge(r, direction, clockwise).collect())
```

Comment:
There's no need to collect with the current set of changes.

Reply:
Then the return value would need to change to `impl Iterator<Item = impl ExactSizeIterator<Item = Self>>`

Reply:
Seems to be working fine